timeseries_storage/blueflood/blueflood.go

```diff
@@ -333,8 +333,8 @@ func (b *Blueflood) constructURL(
 // fetches from the backend. on error, it returns an instance of api.TimeseriesStorageError
 func (b *Blueflood) fetch(request api.FetchTimeseriesRequest, queryUrl *url.URL) (queryResponse, []byte, error) {
 	log.Debugf("Blueflood fetch: %s", queryUrl.String())
-	success := make(chan queryResponse)
-	failure := make(chan error)
```

Comment:
In the event of a timeout, the goroutine spawned to perform the fetch will wait for someone to receive from `success` or `failure`. By giving them capacity, it won't have to wait for someone on the other end.

Otherwise, this will create garbage that Go's garbage collector won't ever touch- the goroutine itself will become eternally frozen.